src/main/java/Database/playlistLibrary.java

```diff
@@ -1,4 +1,181 @@
 package Database;
 
-public class playlistLibrary {
+import Entities.Playlist;
+import Entities.Song;
+import Entities.User;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Playlist database class in the Frameworks & Drivers layer that follows the singleton design pattern, and
+ * implements an interface in the Application Business Rules layer for dependency inversion purposes.
+ * Note: since this is a singleton class, the constructor is private, and it can NOT be created with the
+ * new keyword outside this class.
+ */
+public class playlistLibrary implements playlistAccessInterface {
```

Comment:
think some of this went over my head as I'm not super familiar w the database use case, but: 
- do we care about if multiple users manage a playlist? was not sure that was accounted for in buildPlaylist and AFAIK Nick had a collaborative boolean option in his PR. I guess my question is that will collaborators ever be saved in the playlists.csv file -- if this is the case, then line 83 could possibly throw an error (giving a number of users to users.getUser()). if not, are collaborators stored *only* in the playlist object, and how are we adding collaborators? (in playlist.java the entity, an add collaborator function?) Not the most relevant to our work atm but something to keep in mind possibly. that or just deal with this all later 
- Just to be clear: this singleton manages *all* playlists ever created in our program? as opposed to the collection of playlists one particular user owns/collaborates on. in which case, would we not want RegularUser.java to have a playlistList instance attribute, so that we can present this list to the View eventually on our dashboard?

Reply:
- I was assuming the collaborative definition from Spotify here, in that you can have a collaborative playlist edited by many but it still has one single owner, so line 83 would always be referring to a single user. Saving collaborators separately could be something to look at though, and it can be a simple addition in this code if @NicholasMacasaet ends up implementing it. 
- I probably could’ve explained the design better but the singleton instance of playlistLibrary acts as the single source of persistence for Playlist data. What happens to the Playlist objects once they’re loaded in is up to whoever is implementing a use case - they can choose to pull straight from the database or to store Playlist objects elsewhere. All that matters is that all Playlist objects that will persist in the program are stored by the playlistLibrary instance (and therefore playlists.csv).

Reply:
> think some of this went over my head as I'm not super familiar w the database use case, but:
> 
> * do we care about if multiple users manage a playlist? was not sure that was accounted for in buildPlaylist and AFAIK Nick had a collaborative boolean option in his PR. I guess my question is that will collaborators ever be saved in the playlists.csv file -- if this is the case, then line 83 could possibly throw an error (giving a number of users to users.getUser()). if not, are collaborators stored _only_ in the playlist object, and how are we adding collaborators? (in playlist.java the entity, an add collaborator function?) Not the most relevant to our work atm but something to keep in mind possibly. that or just deal with this all later
> * Just to be clear: this singleton manages _all_ playlists ever created in our program? as opposed to the collection of playlists one particular user owns/collaborates on. in which case, would we not want RegularUser.java to have a playlistList instance attribute, so that we can present this list to the View eventually on our dashboard?

I think we should discuss your first point more because it sounds like adding collaborators would either be its own use case or go under the umbrella of my "playlist curation" use case.

Reply:
>     * I probably could’ve explained the design better, but the singleton instance of playlistLibrary acts as the single source of **persistence** for Playlist data. What happens to the Playlist objects once they’re loaded in is up to whoever is implementing a use case - they can choose to pull straight from the database or to store Playlist objects elsewhere. All that matters is that all Playlist objects that will persist in the program are stored by the playlistLibrary instance (and therefore playlists.csv).

Sure, but then do we not want a playlistList in RegularUser (iirc we removed playlistList from abstract class User because ArtistUser does not have playlists). Not sure if RegularUser was complete -- perhaps Olina can confirm?

Reply:
RegularUser is empty in main right now, afaik. But it would be fine to have a playlist list in user, and it probably makes sense too. It’s up to whoever is implementing the stuff around that code - the singleton design only exists to prevent multiple classes from reading/writing to the persistent database.